internal/state/configuration.go

```diff
@@ -99,6 +101,7 @@ func (b *configBuilder) upsertListener(l *listener) {
 	case v1alpha2.HTTPProtocolType:
 		b.http.upsertListener(l)
 	case v1alpha2.HTTPSProtocolType:
+		b.ssl.listeners = append(b.ssl.listeners, l)
```

Comment:
Here we're tweaking/interfering  with the underlining virtualServerBuilder rather making it doing the work.  Or another way to put it: we're enforcing some internal to virtualServerBuilder constraint outside of virtualServerBuilder.  Imho, this makes the code more brittle. I wonder if we could avoid that?

Reply:
I went back and forth on this. Let me see what I can do.